src/community/mod.rs

```diff
@@ -47,6 +64,104 @@ pub struct WikiPage {
 
 #[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
 #[serde(crate = "near_sdk::serde")]
+pub struct AddOnConfig {
+    pub id: AddOnConfigId,
+    pub parameters: String,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone, PartialEq, Debug)]
+#[serde(crate = "near_sdk::serde")]
+pub struct CommunityAddOn {
+    pub addon_id: AddOnId,
+    pub display_name: String,
+    pub icon: String,
+    pub enabled: bool,
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct AddOn {
```

Comment:
If I add `#[serde(crate = "near_sdk::serde")]` to the AddOn struct and derive Serialize. It doesn't satisfy the requirements fo the Serialize trait because of line 87. (The LookupMap)

Reply:
1. LookupMap is not serializable as you cannot fetch the values from the storage unless you know the key you used to store it there in the first place.
2. It is expensive to fetch all elements of near-sdk collections - those are designed to store thousands records and thus iterating over the whole collection is a foot gun (you would run out of gas), so thanks to the fact that these methods are not implemented, it stopped you here.

Why do you need a LookupMap for 2 elements? This is some overengineering.

Reply:
I already thought so, I had reverted it back to a Vec yesterday.